tests/src/Kernel/OgDeleteOrphanedGroupContentTest.php

```diff
@@ -0,0 +1,212 @@
+    if ($run_cron) {
+      $this->container->get('cron')->run();
+    }
+
+    // Invoke the processing of the orphans.
+    /** @var \Drupal\og\OgDeleteOrphansInterface $plugin */
+    $plugin = $this->ogDeleteOrphansPluginManager->createInstance($plugin_id, []);
+    $plugin->process();
+
+    // Verify the group content is deleted.
+    $this->assertFalse($this->group_content, 'The orphaned node is deleted.');
+
+    // Check that the queue is now empty.
+    $this->assertEquals(0, $queue->numberOfItems());
+  }
+
+  /**
+   * Tests that orphaned content is not deleted when the option is disabled.
+   *
+   * @dataProvider ogDeleteOrphansPluginProvider
+   */
+  function testDisabled($plugin_id) {
+    $this->markTestSkipped();
+    return;
+    // Disable deletion of orphans in the configuration and configure the chosen
+    // plugin.
+    $this->config('og.settings')
+      ->set('delete_orphans', FALSE)
+      ->set('delete_orphans_plugin_id', $plugin_id)
+      ->save();
+
+    // Delete the group.
+    $this->group->delete();
+
+    // Invoke the processing of the orphans.
+    /** @var \Drupal\og\OgDeleteOrphansInterface $plugin */
+    $plugin = $this->ogDeleteOrphansPluginManager->createInstance($plugin_id, []);
+    $plugin->process();
+
+    // Reload the group content that was used during the test.
+    $group_content = Node::load($this->group_content->id());
+
+    // Verify the orphaned node is not deleted.
+    $this->assertTrue($group_content, 'The orphaned node is not deleted.');
+  }
+
+  /**
+   * Provides OgDeleteOrphans plugins for the tests.
+   *
+   * @return array
+   *   An array of test properties. Each property is an indexed array with the
+   *   following items:
+   *   - A string containing the plugin name being tested.
+   *   - A boolean indicating whether or not cron jobs should be run.
+   *   - A string defining the queue that is used by the plugin.
+   */
+  public function ogDeleteOrphansPluginProvider() {
+    return [
+      ['batch', FALSE, 'og_orphaned_group_content'],
+      ['cron', TRUE, 'og_orphaned_group_content_cron'],
+      ['simple', FALSE, 'og_orphaned_group_content'],
+    ];
+  }
+
+  /**
+   * Tests the moving of the node to another group when deleting a group.
+   *
+   * @todo This test doesn't make any sense to me. The way I read it is that if
+   *   multiple groups are present and one of the groups is deleted then its
+   *   content is expected to be moved into a random other group? This seems
+   *   dangerous, it might expose private data.
+   *
+   *   This might be useful for child groups that are related to parent groups,
+   *   so that when a child group is deleted its content will be moved to the
+   *   parent, but then there should be a very clear indication of the parent-
+   *   child relation which is missing in this test. Here the content just moves
+   *   to whatever random group that is available.
+   */
+  function _testMoveOrphans() {
```

Comment:
yeah, I see that there's something missing there -- telling it where to move. I'll have to refresh my memory..